src/buf/buf_impl.rs

```diff
@@ -1393,6 +1396,27 @@ impl<T: Buf + ?Sized> Buf for Box<T> {
     deref_forward_buf!();
 }
 
+impl<P: Buf> Buf for Pin<P>
+    where
+        P: DerefMut + Unpin,
```

Comment:
I think you can remove `P: Unpin` bound here. https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=796867674e9da51d58f8881f829a67a6